### ISSUE TITLE
[Ray AIR/Train] Explain the difference between PyTorch Lightning trainer checkpoint config and Ray AIR checkpoint config

### ISSUE THREAD

**Issue by scottsun94** [ray-project/ray#36314]
### Description

According to @woshiyyya

> The checkpointing process is:
> Lightning save a checkpoint file according to the config in LightningConfigBuilder.checkpointing()
> LightningTrainer reports this new checkpoint to AIR
> AIR saves the AIR checkpoints according to AIR CheckpointConfig

1. We should better document this process in the doc for PyTorch Lightning Trainer. 
2. We should document which parameters in [ray.air.CheckpointConfig](https://docs.ray.io/en/latest/ray-air/api/doc/ray.air.CheckpointConfig.html#ray-air-checkpointconfig) apply to which trainers.

**Comment by scottsun94** [comment]
cc: @matthewdeng @woshiyyya

**Comment by woshiyyya** [comment]
Actually we do have docstring in LightningConfigBuilder that describes this behavior. But perhaps it needs more observability? https://docs.ray.io/en/latest/train/api/doc/ray.train.lightning.LightningConfigBuilder.checkpointing.html#ray-train-lightning-lightningconfigbuilder-checkpointing

**Comment by scottsun94** [comment]
To me, it's a bit hard to understand and it also misses some info (e.g., some parameters in ray.air.configs.CheckpointConfig doesn't apply when using lightningTrainer). 

If I were to rewrite it, here is my 5-min version:

> LightningTrainer creates a subclass instance of the ModelCheckpoint callback with the kwargs. It handles checkpointing and metrics logging logics. This method is **not** a replacement for the ray.air.configs.CheckpointConfig. Specify both lightning checkpointing strategy and AIR checkpoint strategy to properly control the checkpointing behavior. 
>
>Here is how they work together:
> 
> 1. LightningTrainer saves checkpoint files according to the config in LightningConfigBuilder.checkpointing()
> 2. LightningTrainer reports saved checkpoint files to AIR. 
>      - the callback periodically reports the latest metrics and checkpoint to the AIR session via [session.report()](https://docs.ray.io/en/latest/ray-air/api/doc/ray.air.session.report.html#ray.air.session.report). The report frequency matches the checkpointing frequency in LightningConfigBuilder.checkpointing(). You have to make sure that the target metrics (e.g. metrics defined in [TuneConfig](https://docs.ray.io/en/latest/tune/api/doc/ray.tune.TuneConfig.html#ray.tune.TuneConfig) or [CheckpointConfig](https://docs.ray.io/en/latest/ray-air/api/doc/ray.air.CheckpointConfig.html#ray.air.CheckpointConfig)) are ready when a new checkpoint is being saved.
> 3. AIR saves the AIR checkpoints according to AIR CheckpointConfig.
>     - If CheckpointConfig is not provided, AIR stores all the reported checkpoints reported by LightningTrainer by default.
>     - Only ..., ..., ... parameters apply when using LightningTrainer.
>

One thing that I still don't completely understand is `It handles checkpointing and metrics logging logics.` What does "metrics logging" mean here?

**Comment by woshiyyya** [comment]
Yes your thoughts make sense. I am working on unifying the Lightning checkpoint configs and AIR CheckpointConfigs, after that we can make a clearer explanation of the relationships and behavior between them.

> It handles checkpointing and metrics logging logics.

This actually refers to the metrics dict you reported to AIR session (`session.report(metrics, checkpoint)`)

>

**Comment by dstrodtman** [comment]
Closing as obsolete. Ray AIR was deprecated in 2023, and the APIs referenced here have moved into Ray Train (typically with reshaped surface area). If you're hitting a similar gap in the current Train docs, please open a fresh issue with a link to the page.

Context: docs-team triage of the OSS docs issue backlog.
